src/evagram/database/input_tool.py

```diff
@@ -0,0 +1,174 @@
+        self.open_session()
+
+    def open_session(self):
+        # Verifying expected parameters
+        required = {'owner', 'experiment', 'eva_directory'}
+        difference = required.difference(self.wf_dictionary)
+        if len(difference) > 0:
+            print(f"Missing required parameters for workflow dictionary: {difference}")
+            return 1
+        with self.conn:
+            with self.conn.cursor() as cur:
+                cur.execute("SELECT pg_backend_pid();")
+                conn_pid = cur.fetchone()[0]
+                cur.execute("SELECT usename FROM pg_stat_activity WHERE pid=%s", (conn_pid,))
+                conn_username = cur.fetchone()[0]
+                if conn_username != self.wf_dictionary["owner"]:
+                    print("Terminating session, invalid workflow parameters.")
+                    return 1
+                self.owner_id = add_current_user(cur, self.wf_dictionary["owner"])
+                self.experiment_id = add_current_experiment(
+                    cur, self.wf_dictionary["experiment"], self.owner_id)
+                print("Session created successfully! Running task...")
+                self.run_task()
+                print("Task completed!")
+        return 0
+
+    def run_task(self):
+        eva_directory_path = self.wf_dictionary["eva_directory"]
+        with self.conn.cursor() as cur:
+            for observation_dir in os.listdir(eva_directory_path):
+                observation_path = os.path.join(eva_directory_path, observation_dir)
+                if os.path.isdir(observation_path):
+                    for plot in os.listdir(observation_path):
+                        if plot.endswith(".pkl"):
+                            add_plot(
+                                cur, eva_directory_path, observation_dir, plot, self.experiment_id)
+        return 0
+
+
+def insert_table_record(cur, data, table):
+    cur.execute(f"SELECT * FROM {table} LIMIT 0")
+    colnames = [desc[0] for desc in cur.description]
+    # filter data to contain only existing columns in table
+    data = {k: v for (k, v) in data.items() if k in colnames}
+
+    query = f"INSERT INTO {table} ("
+    query += ', '.join(data)
+    query += ") VALUES ("
+    query += ', '.join(["%s" for _ in range(len(data))])
+    query += ")"
+
+    cur.execute(query, tuple(data.values()))
+
+
+def add_current_user(cur, username):
+    # Adds the current user in the workflow and returns its identifier in the database
+    # TODO: perform validation checks on arguments
+    cur.execute("SELECT (owner_id) FROM owners WHERE username=%s", (username,))
+    current_user = cur.fetchall()
+    if len(current_user) == 1:
+        # returns owner_id of current user
+        return current_user[0][0]
+    else:
+        # creates a new user by the specified username
+        user_obj = {"username": username}
+        insert_table_record(cur, user_obj, "owners")
+        return add_current_user(cur, username)
```

Comment:
Why not fetch the user ID after inserting the table record instead of doing another function call? Is it optimal to do the recursion instead?